_posts/2019-02-04-putting-a-plug-app-on-heroku.md

```diff
@@ -0,0 +1,98 @@
+---
+author: Bobby Grayson
+categories: general
+date: 2019-02-03
+layout: post
+title: Getting a Plug App Working With Heroku
+---
+
+In our [previous_post](TODO_LINK) we built a very simple Portfolio app with `Plut.Router`.
+In this (brief) post, we will show how to get things up and running on [Heroku](http://heroku.com) for a vanilla `Plug` app.
+It really is quite easy, but today when I set out to do it, the resources I found either focused on Phoenix or had a few holes, so I figured I would write up a simple one here.
+
+## Let's Do It!
+To deploy to Heroku, your first step is always going to be your `Procfile`.
+The `Procfile` essentially determines what boots up your app.
+Ours is very simple, we will put it at the top level.
+
+```
+web: mix run --no-halt
+```
+
+Thats it!
+
+Now, the reason Heroku is so easy is because they have a _lot_ of built in tooling to make deploying your app easy.
+In the case of Elixir, we need a buildpack that will handle our compilation and to run the app and get the proper dependencies on our server.
+To get any further, you will need a Heroku account and to have their toolbelt installed.
+See [their instructions](https://devcenter.heroku.com/articles/heroku-cli#download-and-install) to get up and running.
+
+Once you have this, run `heroku login`, and then `heroku create`.
+This gets the basics set up for us.
+Now, we can add our buildpack.
+
+```
+heroku buildpacks:set https://github.com/HashNuke/heroku-buildpack-elixir
+```
+
+Now, we need to set our env to prod on heroku:
+
+```
+heroku config:set MIX_ENV=prod
+```
+
+We also need to give our buildpack some config so it knows what our Elixir/Erlang environments are like.
+
+```
+#./elixir_buildpack.config
+erlang_version=21.1
+elixir_version=1.8
+```
+
+Now, we want to set a few configs for heroku:
+
+```
+$ heroku config:set MIX_ENV=prod
+$ heroku config:set PORT=4001
+```
+
+Now that we've set this, let's change our application initialization to use the configured port from the env:
```

Comment:
This note can go away because we're not configuring a port. Instead we should mention that we need to update our `lib/router_example/application.ex` file to fetch the port from the ENV.